script.js

```diff
@@ -1,4 +1,238 @@
+  var randomInteger = Math.floor(randomDecimal);
+  var randomDiceNum = randomInteger + 1;
+  return randomDiceNum;
+};
+
+// Global Variables for Normal Dice Game
+var p1DiceOne = 0;
+var p1DiceTwo = 0;
+var p2DiceOne = 0;
+var p2DiceTwo = 0;
+var player1CombinedNum = 0;
+var player2CombinedNum = 0;
+p1DiceOne = randomDiceRoll();
+p1DiceTwo = randomDiceRoll();
+p2DiceOne = randomDiceRoll();
+p2DiceTwo = randomDiceRoll();
+
+// Global Variables for Variable Dice Game
+var gameMode = 'Enter Mode';
+var numOfDice = 0;
+var playerScore = 0;
+var player1Score = 0;
+var player2Score = 0;
+var totalp1Score = 0;
+var totalp2Score = 0;
+var p1DiceValues = [];
+var p1DicePosition = [];
+var p2DiceValues = [];
+var p2DicePosition = [];
+var dicePosition = [];
+var diceValues = [];
+
+// Helper function to list the indexes in the array
+var getDicePosition = function (diceLength) {
+  var index = 0;
+  while (index < diceLength) {
+    dicePosition.push(index);
+    index = index + 1;
+  }
+  return dicePosition;
+};
+
+// Helper function to list the random dice roll values
+var getNumOfDiceAndRoll = function (diceLength) {
+  var index = 0;
+  while (index < diceLength) {
+    diceValues.push(randomDiceRoll());
+    index = index + 1;
+  } return diceValues;
+};
+
+// Helper function to generate concatenated number after player's index choice
+var generatePlayerScore = function (input) {
+  var positions = input.split('');
+  var playerNumberArray = [];
+  var index = 0;
+  while (index < diceValues.length) {
+    playerNumberArray.push(diceValues[positions[index]]);
+    index = index + 1;
+  }
+  playerScore = playerNumberArray.join('');
+  return playerScore;
+};
+
+// Start Game
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = '';
+  var invalidMessage = 'Please enter a valid position number';
+  // User to choose which mode to play: Normal or Variable Dice Mode
+  if (gameMode == 'Enter Mode') {
+    if (input == 'normal') {
+      myOutputValue = 'You have chosen the normal mode. Click submit to roll dice.';
+      gameMode = 'normal';
+    }
+    else if (input == 'variable') {
+      myOutputValue = 'Please enter the qty of dice you want.';
+      gameMode = 'Player 1 Variable Dice Mode';
+    }
+    else { myOutputValue = 'Please enter "normal" or "variable" to select mode.';
+    }
+    return myOutputValue;
+  }
+
+  // Ask player1 how many dice they want to play with
+  if (gameMode == 'Player 1 Variable Dice Mode') {
+    if (isNaN(Number(input)) == false && input != '') {
+      numOfDice = input;
+      // List the index position of the dice values in order using the helper function
+      // Match it to Player 1
+      dicePosition = getDicePosition(numOfDice);
+      p1DicePosition = dicePosition;
+      // List the dice values using the helper function
+      // Match it to Player 1
+      diceValues = getNumOfDiceAndRoll(numOfDice);
+      p1DiceValues = diceValues;
```

Comment:
Can be shortened to `p1DiceValues = getNumOfDiceAndRoll(numOfDice)`.

`numOfDice` can be more succinct as `numDice`. Could `getNumOfDiceAndRoll` be shortened to `getDiceRolls`?